src/topology/gene.rs

```diff
@@ -239,9 +239,9 @@ where
     T: Float,
 {
     fn cmp(&self, other: &Self) -> Ordering {
-        if self.output.layer == other.output.layer && self.output.index == other.output.index {
+        if self.output.layer == other.output.layer {
             Ordering::Equal
-        } else if self.output.layer < other.output.layer && self.output.layer < other.output.layer {
+        } else if self.output.layer < other.output.layer {
```

Comment:
This one should be

`else if self.output.layer < other.output.layer  || (self.output.layer == other.output.layer && self.output.layer < other.output.layer)`